src/coreComponents/constitutive/diffusion/DiffusionBase.hpp

```diff
@@ -0,0 +1,195 @@
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  /**
+   * @brief Getter for the number of fluid phases
+   * @return the number of fluid phases
+   */
+  integer numFluidPhases() const { return LvArray::integerConversion< integer >( m_phaseNames.size() ); }
+
+  /**
+   * @brief Getter for the phase names
+   * @return an arrayView of phase names
+   */
+  arrayView1d< string const > phaseNames() const { return m_phaseNames; }
+
+  /**
+   * @brief Getter for the phase diffusivity multipliers
+   * @return an arrayView of multipliers
+   */
+  arrayView3d< real64 const > phaseDiffusivityMultiplier() const { return m_phaseDiffusivityMultiplier; }
+
+  /**
+   * @brief Getter for the diffusivities in the subRegion
+   * @return an arrayView of diffusivities
+   */
+  arrayView3d< real64 const > diffusivity() const { return m_diffusivity; }
+
+  /**
+   * @brief Getter for the derivatives of diffusivity wrt temperature in the subRegion
+   * @return an arrayView of derivatives of diffusivities wrt temperature
+   */
+  arrayView3d< real64 const > dDiffusivity_dTemperature() const { return m_dDiffusivity_dTemperature; }
+
+  /**
+   * @brief Initialize the diffusivity state (needed when diffusion depends on temperature)
+   * @param[in] initialTemperature the initial temperature field after reservoir initialization
+   *
+   * Note: this is needed because for now, the temperature is treated **explicitly** in the diffusion tensor
+   */
+  virtual void initializeTemperatureState( arrayView1d< real64 const > const & initialTemperature ) const
+  { GEOS_UNUSED_VAR( initialTemperature ); }
+
+  /**
+   * @brief Save the temperature state (needed when diffusion depends on temperature)
+   * @param[in] convergedTemperature the converged temperature field
+   *
+   * Note: this is needed because for now, the temperature is treated **explicitly** in the diffusion tensor
+   */
+  virtual void saveConvergedTemperatureState( arrayView1d< real64 const > const & convergedTemperature ) const
+  { GEOS_UNUSED_VAR( convergedTemperature ); }
+
+  struct viewKeyStruct : ConstitutiveBase::viewKeyStruct
+  {
+    static constexpr char const * phaseNamesString() { return "phaseNames"; }
+    static constexpr char const * defaultPhaseDiffusivityMultiplierString() { return "defaultPhaseDiffusivityMultipliers"; }
+  };
+
+private:
+
+  /**
+   * @brief Function called internally to resize member arrays
+   * @param size primary dimension (e.g. number of cells)
+   * @param numPts secondary dimension (e.g. number of gauss points per cell)
+   */
+  void resizeFields( localIndex const size, localIndex const numPts );
+
+protected:
+
+  virtual void postProcessInput() override;
+
+  /// phase names read from input
+  string_array m_phaseNames;
+
+  /// array of phase-wise default diffusion multipliers
+  array1d< real64 > m_defaultPhaseDiffusivityMultiplier;
```

Comment:
what's the physical meaning of a diffusivity multiplier? Is it because diffusivity varies by phase? If that's the case why not sizing it based on the number of phases?

Reply:
Yes, the diffusivity varies by phase:

<img width="163" alt="Screenshot 2023-09-18 at 11 34 58 PM" src="https://github.com/GEOS-DEV/GEOS/assets/20196406/3c40eded-dae5-4869-8d7e-ce33ba9e3b9e">

The main reason why it is implemented with 1) a unique diffusion tensor and 2) diffusivity multipliers is the `computeWeights` function here: 
https://github.com/GEOS-DEV/GEOS/blob/39f6373ea92a0501c256960783b5e942c3c3ebfa/src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp#L1064-L1068
Right now I call `computeWeights` only once and then I "scale" it for each phase with the phase multipliers to get the phase diffusivity. 

But we could also remove the phase multipliers and either 
1. Call `computeWeights` `n_p` times (once for each phase)
2. Call `computeWeights` once on a phase-based diffusivity tensor

In both cases, we would need to add a `computeWeights` function that takes a:
```
CoefficientAccessor< arrayView4d< real64 const > > const & coefficient
```
and refactor slightly the existing function to avoid code duplication. But that would be cleaner and also very useful for dispersion which will have the same issue. Let me know what you think, I would be glad to try it out

Reply:
I am also not so sure that we want to use harmonic averaging for the diffusivity so maybe we will wan to do that.